frontends/api/jest.config.ts

```diff
@@ -1,3 +1,4 @@
+/** @jest-config-loader esbuild-register */
```

Comment:
I was getting errors on node ^v22.20 that seemed related to
- https://github.com/jestjs/jest/issues/15682
- https://github.com/nodejs/typescript/issues/37

By default when using TS config files, jest uses `ts-node`, and that has some issue I don't fully understand starting in Node 22.18.

It can be worked around with `NODE_OPTIONS='--no-experimental-strip-types' yarn test`, but that didn't seem great. Supposedly the issue was fixed in Node v24.4, but trying that locally it still was still failing to parse the TS config files.

Switching from ts-node to es-build fixed the issue, and ts-node is kinda dead anyway.

Note: We actually already had esbuild-register installed via storybook, though I added it explicitly as a dev dep now.